api/internal/localizer/localizer.go

```diff
@@ -242,6 +244,48 @@ func (lc *localizer) localizeGenerator(generator *types.GeneratorArgs) (err erro
 	return nil
 }
 
+// localizeHelm localizes helm values files and copies local chart homes.
+func (lc *localizer) localizeHelm(kust *types.Kustomization) error {
+	for i, legacy := range kust.HelmChartInflationGenerator {
+		locFile, err := lc.localizeFileIfSet(legacy.Values)
+		if err != nil {
+			return errors.WrapPrefixf(err, "unable to localize helmChartInflationGenerator entry %d values", i)
+		}
+		kust.HelmChartInflationGenerator[i].Values = locFile
+
+		if legacy.ChartHome != "" {
+			locDir, err := lc.copyChartHome(legacy.ChartHome)
+			if err != nil {
+				return errors.WrapPrefixf(err, "unable to copy helmChartInflationGenerator entry %d", i)
+			}
+			kust.HelmChartInflationGenerator[i].ChartHome = locDir
+		}
+	}
+	for i, chart := range kust.HelmCharts {
+		locFile, err := lc.localizeFileIfSet(chart.ValuesFile)
+		if err != nil {
+			return errors.WrapPrefixf(err, "unable to localize helmCharts entry %d valuesFile", i)
+		}
+		kust.HelmCharts[i].ValuesFile = locFile
+	}
+	if kust.HelmGlobals != nil {
+		if kust.HelmGlobals.ChartHome != "" {
+			locDir, err := lc.copyChartHome(kust.HelmGlobals.ChartHome)
+			if err != nil {
+				return errors.WrapPrefixf(err, "unable to copy helmGlobals")
+			}
+			kust.HelmGlobals.ChartHome = locDir
+		}
+	}
+	if useDefaultChartHome(kust) {
```

Comment:
@natasha41575 As requested, I only localize the default chart home if a chart is present without a home.